processor/src/host/advice/mod.rs

```diff
@@ -0,0 +1,768 @@
+            AdviceInjector::MapValueToStack {
+                include_len,
+                key_offset,
+            } => self.copy_map_value_to_adv_stack(process, *include_len, *key_offset),
+            AdviceInjector::UpdateMerkleNode => {
+                let depth = process.get_stack_item(4);
+                let index = process.get_stack_item(5);
+                let old_root = [
+                    process.get_stack_item(9),
+                    process.get_stack_item(8),
+                    process.get_stack_item(7),
+                    process.get_stack_item(6),
+                ];
+                let new_node = [
+                    process.get_stack_item(13),
+                    process.get_stack_item(12),
+                    process.get_stack_item(11),
+                    process.get_stack_item(10),
+                ];
+                let (path, _) = self.update_merkle_node(old_root, &depth, &index, new_node)?;
+                Ok(HostResponse::MerklePath(path))
+            }
+            AdviceInjector::DivU64 => self.push_u64_div_result(process),
+            AdviceInjector::Ext2Inv => self.push_ext2_inv_result(process),
+            AdviceInjector::Ext2Intt => self.push_ext2_intt_result(process),
+            AdviceInjector::SmtGet => self.push_smtget_inputs(process),
+            AdviceInjector::SmtSet => self.push_smtset_inputs(process),
+            AdviceInjector::SmtPeek => self.push_smtpeek_result(process),
+            AdviceInjector::MemToMap => self.insert_mem_values_into_adv_map(process),
+            AdviceInjector::HdwordToMap { domain } => {
+                self.insert_hdword_into_adv_map(process, *domain)
+            }
+            AdviceInjector::HpermToMap => self.insert_hperm_into_adv_map(process),
+            AdviceInjector::SigToStack { kind } => self.push_signature(process, *kind),
+        }
+    }
+
+    /// Handles the specified advice extractor request.
+    fn get_advice<S: ProcessState>(
+        &mut self,
+        process: &S,
+        advice_extractor: &AdviceExtractor,
+    ) -> Result<HostResponse, ExecutionError> {
+        match advice_extractor {
+            AdviceExtractor::PopStack => self.pop_stack(process).map(HostResponse::Element),
+            AdviceExtractor::PopStackDWord => {
+                self.pop_stack_dword(process).map(HostResponse::DoubleWord)
+            }
+            AdviceExtractor::PopStackWord => self.pop_stack_word(process).map(HostResponse::Word),
+            AdviceExtractor::GetMerklePath => {
+                let depth = process.get_stack_item(4);
+                let index = process.get_stack_item(5);
+                let root = [
+                    process.get_stack_item(9),
+                    process.get_stack_item(8),
+                    process.get_stack_item(7),
+                    process.get_stack_item(6),
+                ];
+                self.get_merkle_path(root, &depth, &index).map(HostResponse::MerklePath)
+            }
```

Comment:
Similar to the previous comment: I wonder if we should create an intermediate method which gets the inputs from the process and then calls `get_merkle_path()`.